std/math/fp4_bb.asm

```diff
@@ -0,0 +1,200 @@
+use std::check::assert;
+use std::check::panic;
+use std::convert::fe;
+use std::convert::int;
+use std::convert::expr;
+use std::math::ff::inv_field;
+use std::prover::eval;
+
+
+/// BETA = 11
+/// NBETA = 2013265910
+
+/// An element of the extension field over the BabyBear Field
+/// relative to the irreducible polynomial X^4 + 11,
+/// where Fp4(a0, a1, a2, a3) is interpreted as a0 + a1 * X + a2 * X^2 + a3 * X^3
+/// T is assumed to either be fe, expr or any other object whose algebraic operations
+/// are compatible with fe.
+enum Fp4<T> {
+    Fp4(T, T, T, T)
+}
+
+/// Converts a base field element to the extension field
+let<T: FromLiteral> from_base: T -> Fp4<T> = |x| Fp4::Fp4(x, 0, 0, 0);
+
+/// Addition for extension field
+let<T: Add> add_ext: Fp4<T>, Fp4<T> -> Fp4<T> = |a, b| match (a, b) {
+    (Fp4::Fp4(a0, a1, a2, a3), Fp4::Fp4(b0, b1, b2, b3)) => Fp4::Fp4(
+        a0 + b0,
+        a1 + b1,
+        a2 + b2,
```

Comment:
just a question, how do we make these addition operation "+" calculated in the BabyBear field? I mean, the result of the sums like a0+b0 ...  are modules babybear prime?

Reply:
When I run the program with this command `cargo run -r -bin powdr pil test_data/std/fp4_bb.asm --field bb -f`, I can specify the field element using `--field` flag. (we currently have bb, gl, bn254). So, the operations are done in the bb